sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/QuickPulseDataSender.java

```diff
@@ -81,10 +82,10 @@ public void run() {
             try {
                 // TODO (harskaur): remove logging when manual testing done
                 logger.verbose("Monitoring point: {}", point.toJsonString());
-                logger.verbose("etag: {}", configuration.getETag());
+                logger.verbose("etag: {}", configuration.get().getETag());
                 Response<CollectionConfigurationInfo> responseMono = liveMetricsRestAPIsForClientSDKs
                     .publishNoCustomHeadersWithResponseAsync(endpointPrefix, instrumentationKey.get(),
-                        configuration.getETag(), transmissionTimeInTicks, dataPointList)
+                        configuration.get().getETag(), transmissionTimeInTicks, dataPointList)
```

Comment:
similar here may be better to call `configuration.get()` above and use the same instance the whole loop iteration